sphinxgallery/gen_rst.py

```diff
@@ -118,133 +78,155 @@ def flush(self):
 
-def extract_docstring(filename, ignore_heading=False):
-    """ Extract a module-level docstring, if any
+    Returns
+    -------
+    blocks : list of (label, content)
+        List where each element is a tuple with the label ('text' or 'code'),
+        and content string of block.
     """
-    lines = open(filename).readlines()
-    start_row = 0
-    if lines[0].startswith('#!'):
-        lines.pop(0)
-        start_row = 1
-    docstring = ''
-    first_par = ''
-    line_iterator = iter(lines)
-    tokens = tokenize.generate_tokens(lambda: next(line_iterator))
-    for tok_type, tok_content, _, (erow, _), _ in tokens:
-        tok_type = token.tok_name[tok_type]
-        if tok_type in ('NEWLINE', 'COMMENT', 'NL', 'INDENT', 'DEDENT'):
+    f = open(source_file)
+
+    blocks = []
+    block = ''
+
+    has_header_string = 0
+    continue_text = False
+    for line in f:
+        # python docstring
+        if line.startswith('"""') and not has_header_string:
+            has_header_string += 1
+            block = line
             continue
-        elif tok_type == 'STRING':
-            docstring = eval(tok_content)
-            # If the docstring is formatted with several paragraphs, extract
-            # the first one:
-            paragraphs = '\n'.join(
-                line.rstrip() for line
-                in docstring.split('\n')).split('\n\n')
-            if paragraphs:
-                if ignore_heading:
-                    if len(paragraphs) > 1:
-                        first_par = re.sub('\n', ' ', paragraphs[1])
-                        first_par = ((first_par[:95] + '...')
-                                     if len(first_par) > 95 else first_par)
-                    else:
-                        raise ValueError("Docstring not found by gallery.\n"
-                                         "Please check the layout of your"
-                                         " example file:\n {}\n and make sure"
-                                         " it's correct".format(filename))
-                else:
-                    first_par = paragraphs[0]
-
-        break
-    return docstring, first_par, erow + 1 + start_row
-
-
-def extract_line_count(filename, target_dir):
-    """Extract the line count of a file"""
-    example_file = os.path.join(target_dir, filename)
-    lines = open(example_file).readlines()
-    start_row = 0
-    if lines and lines[0].startswith('#!'):
-        lines.pop(0)
-        start_row = 1
-    line_iterator = iter(lines)
-    tokens = tokenize.generate_tokens(lambda: next(line_iterator))
-    check_docstring = True
-    erow_docstring = 0
-    for tok_type, _, _, (erow, _), _ in tokens:
-        tok_type = token.tok_name[tok_type]
-        if tok_type in ('NEWLINE', 'COMMENT', 'NL', 'INDENT', 'DEDENT'):
+        # because we only allow for one single docstring
+        elif line.startswith('"""') and has_header_string == 1:
```

Comment:
> You are not getting it right.

OK let's see what reality has to say about that. I tried using plot_strings.py and putting the second `"""` of the docstring in the same line as the last line of text i.e.:

```python
"""
=============================
Constrained Text output frame
=============================

This example captures the stdout and includes it in the
example. If output is too long it becomes automatically
framed into a text area."""
```

This is the output I get:

![plot_strings](https://cloud.githubusercontent.com/assets/1680079/8699210/28c5dcc4-2b04-11e5-9e34-91e43d6bc558.png)

as you can see, the docstring ends up in a code cell rather than a text cell.